.github/workflows/python-test-versions.yml

```
@@ -32,11 +32,19 @@ jobs:
       - name: Install library
         run: poetry install --no-interaction
-      - name: Lint with flake8
+      - name: Lint with ruff (modern replacement for flake8)
         run: |
-          # stop the build if there are Python syntax errors or undefined names
-          poetry run flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics --exclude .venv
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          poetry run flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude .venv
+          # Install ruff for fast, comprehensive linting
+          poetry run pip install ruff
+          # stop the build if there are Python syntax errors or undefined names  
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)